sizeOptimizer.cpp

```diff
@@ -271,44 +417,135 @@ int SizeOptimizer::getInvocationsV8(Polynom &poly, bool isFloat) {
   return result;
 }
 
+int SizeOptimizer::getInvocationsV12(Polynom &poly, bool isFloat) {
+  int result = 0;
+  if (isFloat) {
+    if (poly.value <= V12_REGULAR_MAX_SP)
+    {
+      return 1;
+    }
+    result += getInvocations(V12_RADIX_2_MAX_SP, poly.exponent2);
+    result += getInvocations(V12_RADIX_3_MAX_SP, poly.exponent3);
+    result += getInvocations(V12_RADIX_5_MAX_SP, poly.exponent5);
+    result += getInvocations(V12_RADIX_7_MAX_SP, poly.exponent7);
+    result += getInvocations(V12_RADIX_11_MAX_SP, poly.exponent11);
+  } else {
+    if (poly.value <= V12_REGULAR_MAX_DP)
+    {
+      return 1;
+    }
+    result += getInvocations(V12_RADIX_2_MAX_DP, poly.exponent2);
+    result += getInvocations(V12_RADIX_3_MAX_DP, poly.exponent3);
+    result += getInvocations(V12_RADIX_5_MAX_DP, poly.exponent5);
+    result += getInvocations(V12_RADIX_7_MAX_DP, poly.exponent7);
+    result += getInvocations(V12_RADIX_11_MAX_DP, poly.exponent11);
+  }
+  return result;
+}
+
+
 int SizeOptimizer::getInvocations(Polynom &poly, bool isFloat) {
   switch (version) {
     case (CudaVersion::V_8):
       return getInvocationsV8(poly, isFloat);
-    //	case (CudaVersion::V_9):
-    //		return getInvocationsV9(poly); // FIXME implement
+    case (CudaVersion::V_12):
+      return getInvocationsV12(poly, isFloat);
     default:
       throw std::domain_error("Unsupported version of CUDA");
   }
 }
 
+SizeOptimizer::Polynom SizeOptimizer::SetCorrectValuesToOriginalPolynom(int num, bool isFloat) {
+  Polynom poly;
+  poly.value = num;
+  poly.exponent2 = 0;
```

Comment:
You shouldn't need to do this. Please modify `Polynom` struct with default values and remove this code